app_flutter/lib/task_box.dart

```diff
@@ -234,12 +254,28 @@ class TaskOverlayContents extends StatelessWidget {
                         // TODO(chillers): Open log in new window. https://github.com/flutter/cocoon/issues/436
                       },
                     ),
-                    IconButton(
-                      icon: const Icon(Icons.redo),
-                      onPressed: () {
-                        // TODO(chillers): Rerun task. https://github.com/flutter/cocoon/issues/424
-                      },
-                    ),
+                    if (isDevicelab(task))
+                      ProgressButton(
```

Comment:
I think this widget has reached a point of complexity where it would make sense to split up the buttons into separate classes

Reply:
I think we should rename `TaskOverlayContents` to `TaskOverlayEntry` and fork the actual content into `TaskOverlayContents`. Most of the overhead is from the stuff needed for the `OverlayEntry`.

SG?

Reply:
SGTM!

Reply:
Done.